kernel/objects/execution-context/objects/ExecutionContext.cc

```
@@ -303,10 +342,12 @@ namespace mythos {
       // \TODO don't use PhysPtr because the canonical address belong to the logical addresses
     if (!PhysPtr<void>(fs).canonical() || !PhysPtr<void>(gs).canonical())
       THROW(Error::NON_CANONICAL_ADDRESS);
-    threadState.fs_base = fs;
-    threadState.gs_base = gs;
+    if (state) {
+      state->threadState.fs_base = fs;
+      state->threadState.gs_base = gs;
+    }
```

We might want to return an error code to our unsuspecting user.